RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,613 @@
+      tensor format.
+    - "Dense part" is a strided tensor that is obtained by fixing the
+      indices of all sparse dimensions in a hybrid tensor.
+
+1.  We propose to extend sparse tensor constructors with a keyword
+    argument `fill_value`, used to define the value for
+    unspecified elements of the constructed sparse tensor.
+
+    For instance, the Python signature of `sparse_coo_tensor` would be:
+
+    ```python
+    torch.sparse_coo_tensor(indices, values, size=None, fill_value=None, dtype=None, device=None, requires_grad=False)
+    ```
+
+    where `fill_value=None` indicates the constructor to use the
+    default fill value.
+
+2.  The default fill value is zero.
+
+    This choice is consistent with the interpretation of an unspecified
+    element in Linear Algebra.
+
+    The default fill value of a sparse non-hybrid tensor is a scalar
+    tensor: `torch.tensor(0, dtype=A.dtype, device=A.device)`. For the
+    default fill value of a hybrid tensor, see points 5 and 6 below.
+
+3.  PyTorch functions that have the `layout` argument may use a different
+    `fill_value` when constructing a sparse tensor as defined by the
+    following table:
+
+    | Function              | `fill_value` of returned sparse tensor |
+    | :-------------------- | :----------------------------------- |
+    | `torch.empty`         | undefined value                      |
+    | `torch.empty_like`    | undefined value                      |
+    | `torch.empty_strided` | N/A                                  |
+    | `torch.eye`           | 0                                    |
+    | `torch.full`          | same as `fill_value` argument        |
+    | `torch.full_like`     | same as `fill_value` argument        |
+    | `torch.ones`          | 1                                    |
+    | `torch.ones_like`     | 1                                    |
+    | `torch.zeros`         | 0                                    |
+    | `torch.zeros_like`    | 0                                    |
+
+    Note that this table does not include functions `torch.arange`,
+    `torch.linspace`, `torch.logspace` and `torch.range` that have the
+    `layout` argument as well. We excluded these functions here
+    because these are likely never used for creating sparse tensors.
+    See also point 14.ii below.
+
+4.  The fill value of a sparse tensor can be acquired via the
+    `fill_value()` method that returns a strided `torch.Tensor`
+    instance with the same dtype and storage location properties as
+    the elements of the sparse tensor:
+
+    - `A.fill_value().dtype == A.dtype`
+    - `A.fill_value().device == A.device`
+
+    See point 6 below for how the output of `fill_value()` method is
+    computed.
+
+5.  The fill value of a hybrid sparse tensor has the same shape as the
+    dense part of the tensor.
+
+    For instance, for a sparse tensor `A` in COO storage format we
+    have:
+
+    - `A.fill_value().shape == A.values().shape[1:]`
```

Comment:
For a sparse tensor (not hybrid), `A.fill_value().shape == ()` as it has no dense part. I imagine this applies more generally to hybrid tensors.

Reply:
It applies both for hybrid as well as non-hybrid tensors.

Reply:
In this case, perhaps change "sparse" to "hybrid" above.